src/pgduckdb_userdata_cache.cpp

```diff
@@ -0,0 +1,118 @@
+extern "C" {
+#include "postgres.h"
+
+#include "commands/extension.h"
+#include "foreign/foreign.h"
+#include "utils/inval.h"
+#include "utils/syscache.h"
+}
+
+#include "pgduckdb/pgduckdb_background_worker.hpp"
+#include "pgduckdb/pgduckdb_metadata_cache.hpp"
+#include "pgduckdb/pgduckdb_userdata_cache.hpp"
+#include "pgduckdb/pgduckdb_fdw.hpp"
+
+namespace pgduckdb {
+namespace {
+
+struct {
+	/* First, some metadata for the cache itself */
+	/*
+	 * Does the cache contain valid data, i.e. is it initialized? Or is it
+	 * stale and does it need to be refreshed? If this is false none of the
+	 * other fields should be read.
+	 */
+	bool valid;
+	/*
+	 * An ever increasing counter that is incremented every time the cache is
+	 * revalidated.
+	 */
+	uint64 version;
+
+	/* Anything below is the actual cache content */
+
+	Oid motherduck_foreign_server_oid;
+	Oid motherduck_postgres_role_oid;
+	Oid motherduck_user_mapping_oid;
+} cache = {};
+
+bool callback_is_configured = false;
```

Comment:
nit: should have `static` I think. Otherwise it can conflict with the same named variable somewhere else I think? Similarly for the `cache` global above.

```suggestion
static bool callback_is_configured = false;
```

Reply:
I've moved all of these in an anonymous namespace, so no need to be static.
I was actually wondering if I shouldn't also do the same in the "metadata" cache. WDYT?